File: java/src/main/java/P002.java
@SuppressWarnings("unused")
public class P002 implements Puzzle {
    @Override
    public String solve() {
        long sum = 2;
        long i = 1;
        long j = 2;
        long k = i + j;

        for (; k < 4000000; i = j, j = k, k = i + j) {
            if (k % 2 == 0) {
                sum += k;
            }
        }

        return "" + sum;
    }
}

File: go/src/p002/p002.go
package main

import "fmt"

func main() {
    sum := 0
    for i, j := 1, 2; i < 4000000; i, j = j, j + i {
        if i % 2 == 0 {
            sum += i
        }
    }
    fmt.Println(sum)
}

